homeassistant/components/switchbee/light.py

```diff
@@ -0,0 +1,150 @@
+"""Support for SwitchBee light."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, cast
+
+from switchbee.api import SwitchBeeDeviceOfflineError, SwitchBeeError
+from switchbee.device import ApiStateCommand, DeviceType, SwitchBeeDimmer
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import SwitchBeeCoordinator
+from .entity import SwitchBeeDeviceEntity
+
+MAX_BRIGHTNESS = 255
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _hass_brightness_to_switchbee(value: int) -> int:
+    """Convert hass brightness to SwitchBee."""
+    sb_brightness = int(100 * value / MAX_BRIGHTNESS)
+    # SwitchBee maximum brightness is 99
+    return sb_brightness if sb_brightness != 100 else 99
+
+
+def _switchbee_brightness_to_hass(value: int) -> int:
+    """Convert SwitchBee brightness to hass."""
+    if value == 99:
+        value = 100
+    return round(value * MAX_BRIGHTNESS / 100)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up SwitchBee light."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        SwitchBeeLightEntity(switchbee_device, coordinator)
+        for switchbee_device in coordinator.data.values()
+        if switchbee_device.type == DeviceType.Dimmer
+    )
+
+
+class SwitchBeeLightEntity(SwitchBeeDeviceEntity[SwitchBeeDimmer], LightEntity):
+    """Representation of an SwitchBee light."""
+
+    def __init__(
+        self,
+        device: SwitchBeeDimmer,
+        coordinator: SwitchBeeCoordinator,
+    ) -> None:
+        """Initialize the SwitchBee light."""
+        super().__init__(device, coordinator)
+        self._attr_is_on = False
+        self._attr_brightness = 0
+        self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+        self._attr_available = True
```

Comment:
What is the point of this? True is the default
```suggestion
        self._attr_available = True
```